### PR TITLE
🏗️ DASD-8208 Adding Initial Build YAML and ARM Template

### PR DESCRIPTION
**Pull request by dalemorson**
Adding two files to support the build and release pipelines:
- Adding build yaml file `azure-pipelines.yml` 
- Adding ARM template file `azure/template.json`

### CODE REVIEW

File: azure/template.json [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+            "type": "string"
+        },
+        "sharedKeyVaultName": {
+            "type": "string"
+        },
+        "sharedManagementResourceGroup": {
+            "type": "string"
+        },
+        "loggingRedisConnectionString": {
+            "type": "string"
```
Comment (jack-education) [inline azure/template.json:37]:
```suggestion
            "type": "securestring"
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      arguments: '--configuration $(buildConfiguration) --no-restore'
+      
+  - task: DotNetCoreCLI@2
+    displayName: Unit Tests
+    inputs:
+      command: test
+      projects: '**/*.Application.Tests.csproj'
+      arguments: '--configuration $(buildConfiguration) --no-build'
+
+  - template: azure-pipelines-templates/dependency-check.yml@das-platform-building-blocks
```
Comment (adam230594) [inline azure-pipelines.yml:50]: Can this be changed to `azure-pipelines-templates/build/step/dependency-check.yml`

File: azure/template.json [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+                                "name": "LoggingRedisKey",
+                                "value": "[parameters('loggingRedisKey')]"
+                            },
+                            {
+                                "name": "APPINSIGHTS_INSTRUMENTATIONKEY",
+                                "value": "[reference('api-application-insights').outputs.InstrumentationKey.value]"
+                            },
+                            {
+                                "name": "EnvironmentName",
+                                "value": "[toUpper(parameters('environmentName'))]"
```
Comment (adam230594) [inline azure/template.json:158]: Do they need two config variables with the same value? They'll be using the environmentName config to connect to the storage account so keep the one that is used there

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+- name: BuildConfiguration
+  value: 'release'
+- group: 'Prod Management Resources'
+
+resources:
+  repositories:
+  - repository: das-platform-building-blocks
+    type: github
+    name: SkillsFundingAgency/das-platform-building-blocks
+    ref: refs/heads/master
```
Comment (adam230594) [inline azure-pipelines.yml:24]: You can reference the tags now `refs/tags/0.2.1`

File: azure-pipelines.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    endpoint: 'GitHub (SFA)'
+
+pool:
+  name: 'DAS - Continuous Integration'
+
+steps:
+  - task: DotNetCoreCLI@2
```
Comment (adam230594) [inline azure-pipelines.yml:31]: You can reference the step template we have in platform-building-blocks 

  - template: azure-pipelines-templates/build/step/app-build.yml@das-platform-building-block

File: azure/template.json [hunk 6]
```diff
@@ -0,0 +1,193 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
```
Comment (adam230594) [inline azure/template.json:2]: You'll want `2018-05-01/subscriptionDeploymentTemplate.json#` for subscription ARM template deployments

File: azure/template.json [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+                    "attachedService": {
+                        "value": "[variables('apiAppServiceName')]"
+                    }
+                }
+            }
+        },
+        {
+            "apiVersion": "2017-05-10",
+            "name": "api-app-service",
+            "type": "Microsoft.Resources/deployments",
```
Comment (nbowes24) [inline azure/template.json:118]: As we're not using subscription scope deployments this now needs resourcegroup: "[variables('resourceGroupName')]" adding

File: azure/template.json [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+                    "keyVaultResourceGroup": {
+                        "value": "[parameters('sharedManagementResourceGroup')]"
+                    }
+                }
+            }
+        },
+        {
+            "apiVersion": "2017-05-10",
+            "name": "api-application-insights",
+            "type": "Microsoft.Resources/deployments",
```
Comment (nbowes24) [inline azure/template.json:98]: As we're using subscription scope deployments this now needs resourcegroup: "[variables('resourceGroupName')]" adding

**Pull request merged**